src/pages/my/todo.tsx

```diff
@@ -1,30 +1,108 @@
         'layout.my',
+        'misc.callAssignmentList',
+        'misc.eventList',
         'misc.publicHeader',
         'pages.myTodo',
     ],
 };
 
-export const getServerSideProps : GetServerSideProps = scaffold(async () => {
-    return {
-        props: {},
-    };
+export const getServerSideProps : GetServerSideProps = scaffold(async (context) => {
+    const { user } = context;
+
+    let todoEventsState;
+    let bookedEventsState;
+    let callAssignmentsState;
+    let membershipsState;
+
+    if (user) {
+        await context.queryClient.prefetchQuery('todoEvents', getTodoEvents(context.apiFetch));
+        await context.queryClient.prefetchQuery('bookedEvents', getBookedEvents(context.apiFetch));
+        await context.queryClient.prefetchQuery('callAssignments', getCallAssignments(context.apiFetch));
+        await context.queryClient.prefetchQuery('memberships', getUserMemberships(context.apiFetch));
+
+        todoEventsState = context.queryClient.getQueryState('todoEvents');
+        bookedEventsState = context.queryClient.getQueryState('bookedEvents');
+        callAssignmentsState = context.queryClient.getQueryState('callAssignments');
+        membershipsState = context.queryClient.getQueryState('memberships');
+    }
+
+    if (todoEventsState?.status === 'success'
+        && bookedEventsState?.status === 'success'
+        && callAssignmentsState?.status === 'success'
+        && membershipsState?.status === 'success') {
+
+        return {
+            props: {},
+        };
+    }
+    else {
+        return {
+            notFound: true,
+        };
+    }
 }, scaffoldOptions);
 
 const MyTodoPage : PageWithLayout = () => {
+    const todoEventsQuery = useQuery('todoEvents', getTodoEvents());
+    const bookedEventsQuery = useQuery('bookedEvents', getBookedEvents());
+    const callAssignmentsQuery = useQuery('callAssignments', getCallAssignments());
+    const membershipsQuery = useQuery('memberships', getUserMemberships());
+
+    const { onSignup, onUndoSignup } = useEventResponses();
+
+    if (!todoEventsQuery.data || todoEventsQuery.data.length === 0
+        && !callAssignmentsQuery.data || callAssignmentsQuery.data?.length === 0) {
+        return (
+            <Text data-testid="no-events-placeholder">
+                <Msg id="pages.myTodo.placeholder"/>
+            </Text>
+        );
+    }
+
     return (
-        <Heading level={ 1 }>
-            <Msg id="pages.myTodo.heading"/>
-        </Heading>
+        <>
+            <Heading level={ 1 }>
+                <Msg id="pages.myTodo.heading"/>
+            </Heading>
+            <Heading level={ 2 } marginBottom="0">
+                <Msg id="pages.myTodo.callAssignments"/>
+            </Heading>
+            <Flex marginBottom="0">
+                <CallAssignmentList
+                    callAssignments={ callAssignmentsQuery.data }
+                    memberships={ membershipsQuery.data }
+                />
+            </Flex>
```

Comment:
I'm wondering whether we should render call assignments at all if there are none assigned. An empty list can sometimes be more confusing than no list at all, especially to users who aren't even aware that call assignments are a thing.